ext/fiat/README.mcuboot

```diff
@@ -0,0 +1,13 @@
+This code is based on the boringssl version of fiat-crypto (https://github.com/mit-plv/fiat-crypto)
+
+The version bundled here had most if its non-used code removed, like signing,
+X25519 curve operations not required by verification, etc.
+
+Under boringssl tree, fiat can be found in third_party/fiat. At the time it was
+copied, the last commit was:
+
+commit f109f2087349712d3ac717d15fab48e130618110
```

Comment:
I think this information should be placed in the commit text that adds the fiat code. It doesn't necessarily hurt anything to put it in this `README.mcuboot` file, since it is a good record of the initial version. But it would be good for any commit that brings in changes from the upstream version of this tree have at a minimum the URL and sha hash of the tree version that was brought in.

Another option is to use the `git subtree` command, whose explicit purpose is to bring code from another tree, and keep track of the versions as brought in. The end result wouldn't be different, but it has a specific formatting of the parent URL and hash that allow the tool to later be used to bring patches into the current tree.